### PR TITLE
[0.5.0] Fix query combiner

### PR DESCRIPTION
**Pull request by Disiok**

### CODE REVIEW

File: gpt_index/indices/query/query_combiner/base.py [hunk 1]
```diff
@@ -27,13 +24,17 @@ def __init__(
...
         """Init params."""
         self._index_struct = index_struct
         self._query_transform = query_transform
+        from gpt_index.indices.query.query_runner import QueryRunner
+        assert isinstance(query_runner, QueryRunner)
+        self._query_runner = query_runner
 
     @abstractmethod
-    def run(self, query_obj: BaseGPTIndexQuery, query_bundle: QueryBundle) -> Response:
+    def run(self, query_bundle: QueryBundle, level: int = 0) -> Response:
```
Comment (jerryjliu) [inline gpt_index/indices/query/query_combiner/base.py:37]: Note: i added `arun` in https://github.com/jerryjliu/llama_index/pull/845 in order to support async use cases

File: gpt_index/indices/query/query_runner.py [hunk 2]
```diff
@@ -168,45 +169,13 @@ def _get_query_obj(
...
-        transform_extra_info = {"index_struct": index_struct}
-        query_bundle = query_transform(
-            query_str_or_bundle, extra_info=transform_extra_info
-        )
-
-        # query
+        """This is called via BaseQueryCombiner.run."""
         query_obj = self._get_query_obj(index_struct)
         if self._recursive:
             print(f"Query level : {level} on {index_struct.get_type()}")
```
Comment (jerryjliu) [inline gpt_index/indices/query/query_runner.py:181]: remove print

File: gpt_index/indices/query/query_runner.py [hunk 3]
```diff
@@ -168,45 +169,13 @@ def _get_query_obj(
 
         return query_obj
 
-    def query(
+    def query_transformed(
```
Comment (jerryjliu) [inline gpt_index/indices/query/query_runner.py:172]: i spent like 10 mins thinking about a better way to do this and can't immediately think of one 

we are, however, going to need an async version of this.

File: gpt_index/indices/query/query_runner.py [hunk 4]
```diff
@@ -237,6 +206,50 @@ def query(
...
+            - QueryRunner.query_transformed
+            - BaseQueryCombiner.run
+        
+        QueryRunner.query resolves the current index struct, 
+            then pass control to BaseQueryCombiner.run.
+        BaseQueryCombiner.run applies query transforms and makes multiple queries on the same index.
+        Each query is passed to QueryRunner.query_transformed for execution.
+            During execution, we recursively calls QueryRunner.query if index is a composable graph.
+        """
+        # Resolve index struct from index_id if necessary
```
Comment (jerryjliu) [inline gpt_index/indices/query/query_runner.py:228]: we need an async version of the outer `query` call 

can prob abstract everything except return statement into a helper function

Comment (Disiok) [reply gpt_index/indices/query/query_runner.py:294]: Addressed by https://github.com/jerryjliu/llama_index/pull/929

File: gpt_index/indices/query/query_runner.py (hunk 3 above)
Comment (Disiok) [reply gpt_index/indices/query/query_runner.py:176]: Addressed by https://github.com/jerryjliu/llama_index/pull/929

**Pull request merged**